### DISCUSSION TITLE
`enum-member` vs. `enummember`; `type-parameter`  vs. `typeparameter`

### DISCUSSION THREAD

**Discussion by jdtsmith** [company-mode/company-mode#1369] [Q&A]
I maintain kind-icon, which is used to provide company-kind-based icons in tools like corfu.  A common issue people have noticed is with the `enummember` and `typeparameter` kinds.  Both eglot and lsp-mode specify these names _without dashes_.  But company includes a dash in the kind names it expects for these two (e.g. `enum-member`).  It seems some other completion-kind-providing backends include the dash (see jdtsmith/kind-icon#29).

We need a standard.  Given the name, company's versions should be considered canonical.  In this case, I'd advocate for company to _remove the dashes_, because the [lsp protocol](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#textDocument_documentSymbol) utilizes no dashes for any of its specified kinds:

```
/**
 * A symbol kind.
 */
export namespace [SymbolKind](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#symbolKind) {
	export const File = 1;
	export const Module = 2;
	export const Namespace = 3;
	export const Package = 4;
	export const Class = 5;
	export const Method = 6;
	export const Property = 7;
	export const Field = 8;
	export const Constructor = 9;
	export const Enum = 10;
	export const Interface = 11;
	export const Function = 12;
	export const Variable = 13;
	export const Constant = 14;
	export const String = 15;
	export const Number = 16;
	export const Boolean = 17;
	export const Array = 18;
	export const Object = 19;
	export const Key = 20;
	export const Null = 21;
	export const EnumMember = 22;
	export const Struct = 23;
	export const Event = 24;
	export const Operator = 25;
	export const TypeParameter = 26;
}

export type [SymbolKind](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#symbolKind) = 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10 | 11 | 12 | 13 | 14 | 15 | 16 | 17 | 18 | 19 | 20 | 21 | 22 | 23 | 24 | 25 | 26;
```

**Comment by dgutov** [comment]
> In this case, I'd advocate for company to remove the dashes, because the [lsp protocol](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#textDocument_documentSymbol) utilizes no dashes for any of its specified kinds

But the protocol doensn't use lowercase characters either, does it?

Do you know which values are returned by lsp-mode's and eglot's `:company-kind` functions? With or without the dash?

**Reply by dgutov** [reply]
Anyway, let's move this to an issue. It's not like there's going to be a lot of different answers.
